service/src/intTest/java/uk/nhs/adaptors/gp2gp/gpc/GetGpcStructuredComponentTest.java

```diff
@@ -0,0 +1,136 @@
+        var ehrExtractStatus = EhrExtractStatusTestUtils.prepareEhrExtractStatus();
+        ehrExtractStatusRepository.save(ehrExtractStatus);
+
+        GetGpcStructuredTaskDefinition structuredTaskDefinition = buildValidStructuredTask(ehrExtractStatus);
+        getGpcStructuredTaskExecutor.execute(structuredTaskDefinition);
+
+        var ehrExtractUpdated = ehrExtractStatusRepository.findByConversationId(ehrExtractStatus.getConversationId()).get();
+        assertThatInitialRecordWasUpdated(ehrExtractUpdated, ehrExtractStatus);
+
+        var inputStream = storageConnector.downloadFromStorage(ehrExtractStatus.getConversationId() + GPC_STRUCTURED_FILE_EXTENSION);
+        String storageDataWrapperString = IOUtils.toString(inputStream, StandardCharsets.UTF_8);
+
+        var storageDataWrapper = OBJECT_MAPPER.readValue(storageDataWrapperString, StorageDataWrapper.class);
+        assertThatObjectCreated(storageDataWrapper, ehrExtractUpdated, structuredTaskDefinition);
+    }
+
+    @Test
+    public void When_StructuredTaskRunTwice_Expect_ObjectToBeOverwrittenInStroage() throws IOException {
+        var ehrExtractStatus = EhrExtractStatusTestUtils.prepareEhrExtractStatus();
+        ehrExtractStatusRepository.save(ehrExtractStatus);
+
+        GetGpcStructuredTaskDefinition structuredTaskDefinition1 = buildValidStructuredTask(ehrExtractStatus);
+        getGpcStructuredTaskExecutor.execute(structuredTaskDefinition1);
+
+        GetGpcStructuredTaskDefinition structuredTaskDefinition2 = buildValidStructuredTask(ehrExtractStatus);
+        getGpcStructuredTaskExecutor.execute(structuredTaskDefinition2);
+
+        var ehrExtractUpdated = ehrExtractStatusRepository.findByConversationId(ehrExtractStatus.getConversationId()).get();
+        var inputStream = storageConnector.downloadFromStorage(ehrExtractStatus.getConversationId() + GPC_STRUCTURED_FILE_EXTENSION);
+        String storageDataWrapperString = IOUtils.toString(inputStream, StandardCharsets.UTF_8);
+
+        var storageDataWrapper = OBJECT_MAPPER.readValue(storageDataWrapperString, StorageDataWrapper.class);
+        assertThatObjectCreated(storageDataWrapper, ehrExtractUpdated, structuredTaskDefinition2);
+        assertThat(structuredTaskDefinition1.getTaskId()).isNotEqualTo(ehrExtractUpdated.getGpcAccessStructured().getTaskId());
+    }
+
+    @Test
+    public void When_StructuredTaskPatientNotFoundError_Expect_EhrStatusNotUpdatedAndNotSavedToStorage() {
+        var ehrExtractStatus = EhrExtractStatusTestUtils.prepareEhrExtractStatus();
+        ehrExtractStatusRepository.save(ehrExtractStatus);
+
+        GetGpcStructuredTaskDefinition structuredTaskDefinition1 = buildInvalidNHSNumberStructuredTask(ehrExtractStatus);
+        Exception exception = assertThrows(RuntimeException.class, () -> {
+            getGpcStructuredTaskExecutor.execute(structuredTaskDefinition1);
+        });
+        assertThat(exception.getMessage()).isEqualTo(EXPECTED_ERROR_RESPONSE);
```

Comment:
The test states "Expect_EhrStatusNotUpdatedAndNotSavedToStorage" but these things are not asserted. You need to add these assertions to the test.